nx584/controller.py

```diff
@@ -489,7 +489,7 @@ def get_partition_status(self, partition):
         self._queue.append([0x26, partition - 1])
 
     def set_time(self):
-        now = datetime.datetime.now()
+        now = datetime.datetime.now().astimezone()
```

Comment:
This doesn't actually change the date/time returned, it only augments it with a `tzinfo` object. Since we don't use `now.tzinfo` below, this change doesn't do anything.

Since `datetime.datetime.now()` already _does_ return local time, I wonder if you're using a pi which defaults to UTC and you haven't set the timezone?

Reply:
Regarding time:

I tested it last night, and the code change worked for me.

I ran docker stop pynx584 then docker-compose up and it said:
controller INFO Log event: Time set at 2025-06-13 01:xx:xx
...but it was about 9 PM here. I changed the code and stopped and started it again and it et the correct time. I verified by going to the panel.

Today, I reverted the change with a git reset and restarted the container and it still set the correct time. I'm confused, but you know much more about Python than I do, so I defer to your judgement.

Reply:
Regarding time:

root@pynx584:~# ls -l /etc/localtime
lrwxrwxrwx 1 root root 36 Feb 18  2024 /etc/localtime -> /usr/share/zoneinfo/America/New_York
root@pynx584:~# date
Fri Jun 13 20:18:26 EDT 2025

I tested it last night, and the code change worked for me.

I ran docker stop pynx584 then docker-compose up and it said:
controller INFO Log event: Time set at 2025-06-13 01:xx:xx
...but it was about 9 PM here. I changed the code and stopped and started it again and it et the correct time. I verified by going to the panel.

Today, I reverted the change with a git reset and restarted the container and it still set the correct time. I'm confused, but you know much more about Python than I do, so I defer to your judgement.